File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/PlayerConnectionService.java
package top.likoslupus.cellulosesz.api.player;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;
import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult;
import top.likoslupus.cellulosesz.api.text.RichText;

public interface PlayerConnectionService {

    PlatformResult<Void> disconnect(CellPlayer player, RichText reason);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/PlayerConnectionService.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult
import top.likoslupus.cellulosesz.api.text.RichText

public interface PlayerConnectionService {

    public fun disconnect(player: CellPlayer, reason: RichText): PlatformResult<Void>

}
